### DISCUSSION TITLE
Possible bug with linker?

### DISCUSSION THREAD

**Discussion by Pokeman2003** [open-watcom/open-watcom-v2#1359] [General]
Okay, so I've made a really complicated batch make script. Surprisingly despite my complete inexperience, it seems to mostly work, up until it gets to the linker.
[Here is the script and the test bed.](https://github.com/user-attachments/files/18203137/Source.zip) Keep in mind this is very much trial and error, so there's a couple things left to iron out that I definitely haven't noticed yet.
My inputs to the script are as follows:
`%WATCOM%\binnt\wcc386.exe`
`%WATCOM%\binnt\wlink.exe`
`%cd%\CompileFolder` THIS NEEDS TO BE AN EXISTING FOLDER.
`%WATCOM%\h;%WATCOM%\h\nt`

Now, let me walk you through how the linker script is formed. You can find this under the section :link. The first part of the script is immutable and looks like this currently: 
```
system nt_win
option map
name output
```
Outside of nt_win supposedly not existing which is [very weird](https://open-watcom.github.io/open-watcom-v2-wikidocs/lguide.pdf#page=17), there are no issues with this section. The problem arises when I first try to use the autogenerated file list. This list is generated with a for loop, and every file is presented in 8.3 shorthand. In the current file incarnation, it looks like this:
```
file "W:\Project\COMPIL~1\Backend\File\Ingress.obj"
file "W:\Project\COMPIL~1\Backend\LOCALI~1\LocalEN.obj"
file "W:\Project\COMPIL~1\Backend\LOCALI~1\LOCALF~1.OBJ"
file "W:\Project\COMPIL~1\Windows\INITIA~1.OBJ"
```
This on paper would work. These objects absolutely exist. Hell, the command that's used to get these objects, `dir /b /s *.obj`, can ONLY list existing .obj files. However, the linker decided that they do not exist, given this output:
```
Open Watcom Linker Version 2.0 beta Dec  2 2024 02:29:37 (32-bit)
Copyright (c) 2002-2024 The Open Watcom Contributors. All Rights Reserved.
Portions Copyright (c) 1985-2002 Sybase, Inc. All Rights Reserved.
Source code is available under the Sybase Open Watcom Public License.
See https://github.com/open-watcom/open-watcom-v2#readme for details.
Warning! W1107: file W:\Project\COMPIL~1\link: line(1): undefined system name: nt_win
loading object files
Error! E2008: cannot open "W:\Project\COMPIL~1\Backend\File\Ingress.obj" : No such file or directory
Error! E2008: cannot open "W:\Project\COMPIL~1\Backend\LOCALI~1\LocalEN.obj" : No such file or directory
Error! E2008: cannot open "W:\Project\COMPIL~1\Backend\LOCALI~1\LOCALF~1.OBJ" : No such file or directory
Error! E2008: cannot open "W:\Project\COMPIL~1\Windows\INITIA~1.OBJ" : No such file or directory
```
I have no idea what the hell is happening and I'm not sure whether to file a bug report or not, because it could literally just be my incompetence.

**Comment by Pokeman2003** [comment]
Oh it's my incompetence. The quotation marks are setting it off. That probably shouldn't happen, but whatever.

**Comment by jmalak** [comment]
Sorry, you have incorrectly configured Open Watcom
Why you not use correct configuration and next you report issue?
Please read standard readme file where is describe how to configure Open Watcom.
Following environment variable must be setup with regards to your host and target OS
- WATCOM
-  PATH
- INCLUDE

why you use absolute path instead tools name only?
I don't understand why people automaticaly expect that mistake is in OW even if not read documentation and don't configure OW at all.
